src/main/java/com/mongodb/jdbc/BsonTypeInfo.java

```diff
@@ -1,342 +1,230 @@
+    BSON_UNDEFINED("undefined", BsonType.UNDEFINED, Types.OTHER, false, 0, 0, 0, null, null, null),
+    BSON_OBJECTID("objectId", BsonType.OBJECT_ID, Types.OTHER, false, 0, 0, 0, 24, null, null),
+    BSON_BOOL("boolean", BsonType.BOOLEAN, Types.BIT, false, 0, 0, 0, 1, null, 1),
+    BSON_DATE("date", BsonType.DATE_TIME, Types.TIMESTAMP, false, 0, 3, 0, 24, 3, 8),
+    BSON_NULL("null", BsonType.NULL, Types.NULL, false, 0, 0, 0, null, null, null),
+    BSON_REGEX("regex", BsonType.REGULAR_EXPRESSION, Types.OTHER, true, 0, 0, 0, null, null, null),
+    BSON_DBPOINTER("dbPointer", BsonType.DB_POINTER, Types.OTHER, false, 0, 0, 0, null, null, null),
+    BSON_JAVASCRIPT(
+            "javascript", BsonType.JAVASCRIPT, Types.OTHER, true, 0, 0, 0, null, null, null),
+    BSON_SYMBOL("symbol", BsonType.SYMBOL, Types.OTHER, true, 0, 0, 0, null, null, null),
+    BSON_JAVASCRIPTWITHSCOPE(
+            "javascriptWithScope",
+            BsonType.JAVASCRIPT_WITH_SCOPE,
+            Types.OTHER,
+            true,
+            0,
+            0,
+            0,
+            null,
+            null,
+            null),
+    BSON_INT("int", BsonType.INT32, Types.INTEGER, false, 0, 0, 2, 10, null, 4),
+    BSON_TIMESTAMP("timestamp", BsonType.TIMESTAMP, Types.OTHER, false, 0, 0, 0, null, null, null),
+    BSON_LONG("long", BsonType.INT64, Types.BIGINT, false, 0, 0, 2, 19, null, 8),
+    BSON_DECIMAL("decimal", BsonType.DECIMAL128, Types.DECIMAL, false, 34, 34, 10, 34, 34, 16),
+    BSON_MINKEY("minKey", BsonType.MIN_KEY, Types.OTHER, false, 0, 0, 0, null, null, null),
+    BSON_MAXKEY("maxKey", BsonType.MAX_KEY, Types.OTHER, false, 0, 0, 0, null, null, null),
+    BSON_BSON("bson", BsonType.UNDEFINED, Types.OTHER, false, 0, 0, 0, null, null, null);
+
+    private static final Set<String> BSON_TYPE_NAMES =
+            new HashSet<>(
+                    Arrays.asList(
+                            BSON_DOUBLE.bsonName,
+                            BSON_STRING.bsonName,
+                            BSON_OBJECT.bsonName,
+                            BSON_ARRAY.bsonName,
+                            BSON_BINDATA.bsonName,
+                            BSON_UNDEFINED.bsonName,
+                            BSON_OBJECTID.bsonName,
+                            BSON_BOOL.bsonName,
+                            BSON_DATE.bsonName,
+                            BSON_NULL.bsonName,
+                            BSON_REGEX.bsonName,
+                            BSON_DBPOINTER.bsonName,
+                            BSON_JAVASCRIPT.bsonName,
+                            BSON_SYMBOL.bsonName,
+                            BSON_JAVASCRIPTWITHSCOPE.bsonName,
+                            BSON_INT.bsonName,
+                            BSON_TIMESTAMP.bsonName,
+                            BSON_LONG.bsonName,
+                            BSON_DECIMAL.bsonName,
+                            BSON_MINKEY.bsonName,
+                            BSON_MAXKEY.bsonName));
```

Comment:
Missing BSON_BSON type

```suggestion
                            BSON_BSON.bsonName,
                            BSON_MAXKEY.bsonName));
```

Reply:
The "bson" type isn't a real bson type so it was intentionally omitted from this list! I'll add a comment to the code explaining that.

This set is specifically used for the `getBsonTypeInfoByName` method which should fail on unknown type names. The interesting thing to note here is that we have this extra "bson type" called "bson" that is meant to represent "any". It is a construct that only exists in our codebase.